src/megaclient.cpp

```diff
@@ -909,10 +909,18 @@ void MegaClient::exec()
 
                         // notify app in case some attributes were not returned, then redispatch
                         fc->failed(this);
+                        fc->req.disconnect();
+                        fc->req.status = REQ_PREPARED;
+                        fc->timeout.reset();
```

Comment:
Keep the state of the request consistent, reset the timeout and ensure that the request is detached from the network layer. The state of this request is also used to control the state of another request sent using `pendingcs`, so it's important to keep the status updated and the timeout inactive when it isn't needed to prevent a high CPU usage in some circumstances.